File: Python/Problems/Problems/problem1887.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set
from collections import Counter

from .problem import Problem


class Problem1887(Problem):
    def test(self) -> bool:
        nums = [1, 1, 2, 2, 3]

        expected = 4

        result = self.reductionOperations(nums)

        print(result)

        return result == expected

    def reductionOperations(self, nums: List[int]) -> int:
        n = len(nums)
        nums.sort()
        result = 0
        for i in range(n - 2, -1, -1):
            if nums[i] != nums[i + 1]:
                result += n - i - 1

        return result

File: C++/Problems/Problems/problem_1887.cpp
#include "problem_1887.h"

#include <algorithm>

bool problem_1887::test() {
    std::vector nums { 1, 1, 2, 2, 3 };

    int expected = 4;

    auto result = reductionOperations(nums);

    std::cout << result << std::endl;

    return result == expected;
}

int problem_1887::reductionOperations(std::vector<int> &nums) {
    int n = nums.size();
    std::sort(nums.begin(), nums.end());
    int result = 0;
    for (int i = n - 2; i >= 0; --i)
    {
        if (nums[i] != nums[i + 1])
            result += n - i - 1;
    }

    return result;
}
